subsys/bluetooth/host/hci_core.c

```diff
@@ -1254,6 +1254,19 @@ static inline bool rpa_timeout_valid_check(void)
 #endif
 }
 
+static bool rpa_is_new(void)
+{
+#if defined(CONFIG_BT_PRIVACY)
+	/* RPA is considered new if there is less than half a second since the
+	 * timeout was started.
+	 */
+	return k_delayed_work_remaining_get(&bt_dev.rpa_update) >
+	       (RPA_TIMEOUT - K_MSEC(500));
```

Comment:
This will need to be fixed almost as soon as it's merged, since with latest upstream when `CONFIG_LEGACY_TIMEOUT_API` is not enabled both `RPA_TIMEOUT` and `K_MSEC()` are structs whereas the return value of `k_delayed_work_remaining_get()` is `u32_t`. In fact, why not make this immediately the "right way" so that it works with the new `k_timeout_t`-based world?

Reply:
From what I can see both RPA_TIMEOUT and K_MSEC(500) could just be kept in milliseconds, which would solve this.

Reply:
..that said I guess modifying RPA_TIMEOUT type has a trickle effect, so I'm starting to wonder if a follow-up PR wouldn't make more sense. I'll let you choose :)

Reply:
@jhedberg I think the best approach is to do this in a follow-up PR after this and https://github.com/zephyrproject-rtos/zephyr/pull/23911.